src/lib/mina_base/party.ml

```diff
@@ -965,12 +973,13 @@ module Predicate = struct
       ~back:(fun s -> Full s)
 end
 
-module Predicated = struct
+module Preconditioned = struct
```

Comment:
If I didn't already know what this was, I think the term `Preconditioned` might be confusing. `With_precondition` might be clearer, or -- better still -- moving `account_precondition` into body and deleting this type.

Reply:
Yeah, agreed. I'd prefer moving it into body alongside protocol state precondition. Mind if I do it in a different PR?